test/src/test/java/jenkins/model/TransientActionFactoryTest.java

```diff
@@ -147,6 +149,31 @@
         }
     }
 
+    @Issue("JENKINS-51584")
+    @Test
+    public void transientActionsAreNotPersistedOnQueueItems() throws Exception {
+        FreeStyleProject p = r.createFreeStyleProject();
+        FreeStyleBuild build = r.buildAndAssertSuccess(p);
+        // MyProminentProjectAction is only added via the TransientActionFactory and should never be persisted.
+        assertThat(build.getActions(MyProminentProjectAction.class), hasSize(0));
```

Comment:
his is the wrong call as it will return things that are persisted (getAction) as well as actions from the `TransientActionFactory`.  Still, there are 2 entries which is still wrong :)